### PR TITLE
`feat/chain_simulator`

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- WIP
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/node/config/testKeys/validatorKey.pem [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-----BEGIN PRIVATE KEY for 75907bf66c64949f8d1dd003ed8fd815e8dec189a9e50e4bfd7e2592cfdc92641baf500e3533820e0f58c4ed3f39c01750f0017726731fe5bc66a0ab6fc9a7b5661d163e72f099fe29df16fd59ded198dc1423575be1b6e2160e0d84a5ece08a-----
```
Comment (iulianpascalau) [inline cmd/node/config/testKeys/validatorKey.pem:1]: do we still need these?

File: node/external/timemachine/fee/memoryFootprint/memory_test.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ func TestFeeComputer_MemoryFootprint(t *testing.T) {
 	}
 
 	numEpochs := 10000
-	maxFootprintNumBytes := 50_000_000
+	maxFootprintNumBytes := 60_000_000
```
Comment (iulianpascalau) [inline node/external/timemachine/fee/memoryFootprint/memory_test.go:22]: do we have a possible leak that we require this change?

**Review by iulianpascalau — COMMENTED** [review]
**no production components were touched by this feat**

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by schimih — APPROVED** [review]

File: cmd/node/config/testKeys/validatorKey.pem (hunk 1 above)
Comment (miiu96) [reply cmd/node/config/testKeys/validatorKey.pem]: removed

File: node/external/timemachine/fee/memoryFootprint/memory_test.go (hunk 2 above)
Comment (miiu96) [reply node/external/timemachine/fee/memoryFootprint/memory_test.go]: reverted this change

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
please fix the code coverage github action (maybe skip tests for chain simulator if the tests are ran with `-short`)

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Pull request merged**